pkg/common/cns-lib/volume/manager.go

```diff
@@ -248,6 +257,20 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		log.Error(msg)
 		return nil, errors.New(msg)
 	}
+	var datastoreURL string
+	volumeCreateResult := interface{}(taskResult).(*cnstypes.CnsVolumeCreateResult)
+	if volumeCreateResult.PlacementResults != nil {
+		datastoreMoRef := volumeCreateResult.PlacementResults[0].Datastore
+		var dsMo mo.Datastore
+		pc := property.DefaultCollector(m.virtualCenter.Client.Client)
+		err := pc.RetrieveOne(ctx, datastoreMoRef, []string{"summary"}, &dsMo)
+		if err != nil {
+			log.Errorf("Failed to retrieve datastore summary property: %v", err)
```

Comment:
don't we need to fail volume creation if we are not able to retrieve the datastore URL from placement result?

Reply:
Yes. Fixed.